src/components/Profile/index.tsx

```diff
@@ -71,6 +70,23 @@ export const Profile = () => {
     const [modalVisible, setModalVisible] = useState(false)
     const [modalType, setModalType] = useState<'Boost' | 'Invites' | undefined>(undefined)
 
+    // Calculate the number of items that can be displayed on the page
+    const calculateItemsPerPage = () => {
+        const itemHeight = 100 // estimated height of each item
+        const availableHeight = window.innerHeight - 300 // estimated height of the header and footer
+        const calculatedItemsPerPage = Math.floor(availableHeight / itemHeight)
+        setItemsPerPage(calculatedItemsPerPage > 0 ? calculatedItemsPerPage : 1)
+    }
+
+    useEffect(() => {
+        calculateItemsPerPage()
+        window.addEventListener('resize', calculateItemsPerPage)
+
+        return () => {
+            window.removeEventListener('resize', calculateItemsPerPage)
+        }
+    }, [])
+
```

Comment:
Thought: I think this isn't a _great_ approach, because:

- if user zooms, it changes how many items it shows

However, i can't think of anything significantly better

One suggestion though: add some buffer
```suggestion
    // Calculate the number of items that can be displayed on the page
    const calculateItemsPerPage = () => {
    const itemHeight = 100
    const availableHeight = window.innerHeight - 300
    const calculatedItems = Math.floor(availableHeight / itemHeight)
    // Only update if the change is significant
    if (Math.abs(calculatedItems - itemsPerPage) > 1) {
        setItemsPerPage(Math.max(calculatedItems, 1))
    }
}
    useEffect(() => {
        calculateItemsPerPage()
        window.addEventListener('resize', calculateItemsPerPage)
        return () => {
            window.removeEventListener('resize', calculateItemsPerPage)
        }
    }, [])

```

Reply:
I thought this was what you suggested, having x amount per screen size? Sure i'll add buffer